eth-providers/src/base-provider.ts

```diff
@@ -169,6 +188,48 @@ export abstract class BaseProvider extends AbstractProvider {
     this.api.rpc.chain.subscribeFinalizedHeads(async (header: Header) => {
       this._cache!.handleFinalizedBlock(header.number.toNumber());
     }) as unknown as void;
+
+    // TODO: factor this out or maybe rename it to start subscription?
+    this.api.rpc.chain.subscribeNewHeads(async (header: Header) => {
+      if (this._listeners[NEW_HEADS]?.length > 0) {
```

Comment:
why not merge this into the other `subscribeNewHeads`? we will want to avoid multiple subscriptions to reduce load on the node side